tests/components/konnected/test_panel.py

```diff
@@ -551,3 +551,85 @@ async def test_default_options(hass, mock_panel):
             },
         ],
     }
+
+
+async def test_connect_retry(hass, mock_panel):
+    """Test that we create a Konnected Panel and save the data."""
+    device_config = config_flow.CONFIG_ENTRY_SCHEMA(
+        {
+            "host": "1.2.3.4",
+            "port": 1234,
+            "id": "112233445566",
+            "model": "Konnected Pro",
+            "access_token": "11223344556677889900",
+            "default_options": config_flow.OPTIONS_SCHEMA({config_flow.CONF_IO: {}}),
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected",
+        title="Konnected Alarm Panel",
+        data=device_config,
+        options={},
+    )
+    entry.add_to_hass(hass)
+
+    # fail first 2 attempts, and succeed the third
+    mock_panel.get_status.side_effect = [
+        mock_panel.ClientError,
+        mock_panel.ClientError,
+        {
+            "hwVersion": "2.3.0",
+            "swVersion": "2.3.1",
+            "heap": 10000,
+            "uptime": 12222,
+            "ip": "192.168.1.90",
+            "port": 9123,
+            "sensors": [],
+            "actuators": [],
+            "dht_sensors": [],
+            "ds18b20_sensors": [],
+            "mac": "11:22:33:44:55:66",
+            "model": "Konnected Pro",
+            "settings": {},
+        },
+    ]
+
+    # setup the integration and inspect panel behavior
+    with patch.object(hass.helpers.event, "async_call_later") as mock_call_later:
```

Comment:
It's better to not mock call_later and instead fire a time changed event, with our test helper:
https://github.com/home-assistant/core/blob/af5cb948a0396d2031851115dbc848029b2c57b1/tests/common.py#L288

We're not interested in what makes the retry happen, just that it happens after the correct time.

Reply:
Awesome - I didn't realize that helper exists. Will switch to it